ui/app/components/user-preferenced-currency-input/user-preferenced-currency-input.container.js

```diff
@@ -0,0 +1,13 @@
+import { connect } from 'react-redux'
```

Comment:
I wonder if either `user-preferenced-currency-input.container.js` or `currency-input.container.js` can be eliminated?

They both only get a small amount of data from state, and the `currency-input.container.js` is only ever imported in the `user-preferenced-currency-input.component.js`.

It seems like `CurrencyInput` could be an unconnected component that just receives its props from its parent. I think this would make it easier to understand the relation between the `UserPreferencedCurrencyInput` and the `CurrencyInput`. Also could make `CurrencyInput` more flexible for future use.

Reply:
The way I look at this is that `UserPreferencedCurrencyInput` builds on top of `CurrencyInput`, which builds on top of `UnitInput`, the unconnected component. Everything that `UserPreferencedCurrencyInput` and `CurrencyInput` does can be done with `UnitInput` itself, but they're just added layers of convenience and do require being connected to the redux store. It's a much easier API to use for the developer when the component requires just a few props to use and reduces duplication of managing these props.